go/arrow/cdata/cdata.go

```diff
@@ -613,6 +639,12 @@ func (imp *cimporter) checkNumBuffers(n int64) error {
 func (imp *cimporter) importBuffer(bufferID int, sz int64) *memory.Buffer {
 	// this is not a copy, we're just having a slice which points at the data
 	// it's still owned by the C.ArrowArray object and its backing C++ object.
+	if imp.cbuffers[bufferID] == nil {
+		if sz != 0 {
+			panic("invalid buffer")
```

Comment:
"panic" bubbles a regular catchable error, or it's a controlled crash?

Reply:
`panic` *can* be caught and handled via `defer` and `recover`, otherwise it will be a controlled crash with a stack trace (unless debug info was stripped when building).

Reply:
Can't you return a regular error then?

Reply:
sure, had to propagate the error through to the callers since this method previously didn't return an error. latest commit updates this with the error instead of panic.